modules/core/05-port/keeper/keeper.go

```diff
@@ -87,3 +89,24 @@ func (k *Keeper) LookupModuleByPort(ctx sdk.Context, portID string) (string, *ca
 func (k *Keeper) Route(module string) (types.IBCModule, bool) {
 	return k.Router.GetRoute(module)
 }
+
+// AppRoute returns an ordered list of IBCModule callbacks for a given module name, and a boolean indicating
+// whether or not the callbacks are present.
+func (k *Keeper) AppRoute(module string) ([]types.IBCModule, bool) {
+	routes, ok := k.AppRouter.GetRoute(module)
+	if ok {
+		return routes, true
+	}
+
+	for _, prefix := range k.Keys() {
+		if strings.Contains(module, prefix) {
+			return k.AppRouter.GetRoute(prefix)
+		}
+	}
```

Comment:
lets make this well documented since the implications are subtle cc @damiannolan